### PR TITLE
[tvOS] Fix cropped Now Playing image

### PR DESCRIPTION
**Pull request by sy6sy2**
## Description

If the image used for the NowPlaying artwork is not a perfect square, then this image is cropped on the NowPlaying view of the Apple TV (see screenshot).
To fix this issue we can put this image in a square.

## Motivation and Context

Just a little fix for a beautiful NowPlaying image.

## How Has This Been Tested?

On Apple TV 4K NowPlaying view and iPhone 7 NowPlaying view.

## Screenshots (if appropriate):

**Before:**

Capture d’écran 2020-01-13 à 08 55 44

**After:**

Capture d’écran 2020-01-13 à 08 56 32

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by SylvainCecchetto** [comment]
> Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

should it be better now?

**Comment by Rechi** [comment]
It's better, but still not correct.

**Comment by SylvainCecchetto** [comment]
I hope I found the line guilty now ;-)
(I thought `git-clang-format` does more than that by itself ...)

**Comment by SylvainCecchetto** [comment]
Jenkins build this please

File: xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm [hunk 1]
```diff
@@ -49,8 +49,21 @@ - (void)onPlay:(NSDictionary*)item
   {
     if (auto image = [UIImage imageWithContentsOfFile:thumb])
     {
+      auto square_size = image.size.height;
+      if (image.size.width > square_size)
+      {
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: remove brackets for single-line conditions

File: xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm [hunk 2]
```diff
@@ -49,8 +49,21 @@ - (void)onPlay:(NSDictionary*)item
...
+        square_size = image.size.width;
+      }
+      UIImageView* view =
+          [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, square_size, square_size)];
+      [view setImage:image];
+      view.contentMode = UIViewContentModeScaleAspectFit;
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: don't create dummy view for this, use `AVMakeRectWithAspectRatioInsideRect` function to get image drawing rect instead. And then just call `[image drawInRect:]`.

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm (hunk 2 above)
Comment (kambala-decapitator) [reply xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: I guess should be this:
```objc
auto rect = CGRectMake(0, 0, square_size, square_size);
UIGraphicsBeginImageContextWithOptions(rect.size, YES, 0.0);
[image drawInRect:AVMakeRectWithAspectRatioInsideRect(image.size, rect)];
image = UIGraphicsGetImageFromCurrentImageContext();
UIGraphicsEndImageContext();
```

Comment (SylvainCecchetto) [reply xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: Thank you for your help :-)

Comment (kambala-decapitator) [reply xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: I hope you have runtime-tested the change? Because I'm not 100% sure :)

Comment (SylvainCecchetto) [reply xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: Yes I did it, it works ;-)

File: xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm [hunk 3]
```diff
@@ -49,8 +50,16 @@ - (void)onPlay:(NSDictionary*)item
...
+      if (image.size.width > square_size)
+        square_size = image.size.width;
+      auto rect = CGRectMake(0, 0, square_size, square_size);
+      UIGraphicsBeginImageContextWithOptions(rect.size, YES, 0.0);
+      [image drawInRect:AVMakeRectWithAspectRatioInsideRect(image.size, rect)];
+      image = UIGraphicsGetImageFromCurrentImageContext();
+      UIGraphicsEndImageContext();
       if (auto mArt =
-          [[MPMediaItemArtwork alloc] initWithBoundsSize:image.size
+          [[MPMediaItemArtwork alloc] initWithBoundsSize:CGSizeMake(square_size, square_size)
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm]: `rect.size` ;)

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

**Review by kambala-decapitator — APPROVED** [review]

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by SylvainCecchetto** [comment]
Jenkins build this please

**Comment by SylvainCecchetto** [comment]
Jenkins build this please

**Pull request merged**